Content.Shared/Hands/EntitySystems/SharedHandsSystem.Drop.cs

```diff
@@ -122,18 +125,23 @@ public bool TryDrop(EntityUid uid, Hand hand, EntityCoordinates? targetDropLocat
         var userXform = Transform(uid);
         var isInContainer = ContainerSystem.IsEntityOrParentInContainer(uid, xform: userXform);
 
+        // drop the item inside the container if the user is in a container
+        // otherwise, remove the item from their hands and place it at the calculated interaction range position
         if (targetDropLocation == null || isInContainer)
         {
-            // If user is in a container, drop item into that container. Otherwise, attach to grid or map.
             TransformSystem.DropNextTo((entity, itemXform), (uid, userXform));
             return true;
         }
+        else
+        {
+            DoDrop(uid, hand, doDropInteraction: doDropInteraction, handsComp);
 
-        var (itemPos, itemRot) = TransformSystem.GetWorldPositionRotation(entity);
-        var origin = new MapCoordinates(itemPos, itemXform.MapID);
-        var target = targetDropLocation.Value.ToMap(EntityManager, TransformSystem);
-        TransformSystem.SetWorldPositionRotation(entity, GetFinalDropCoordinates(uid, origin, target), itemRot);
-        return true;
+            var (itemPos, itemRot) = TransformSystem.GetWorldPositionRotation(entity);
+            var origin = new MapCoordinates(itemPos, itemXform.MapID);
+            var target = TransformSystem.ToMapCoordinates(targetDropLocation.Value);
+            TransformSystem.SetWorldPositionRotation(entity, GetFinalDropCoordinates(uid, origin, target), itemRot);
+            return true;
+        }
```

Comment:
Shouldn't you remove the else block (keeping the code).
or move the return statements after the if-else blocks?

Reply:
it's functionally no different, it always go to either the if or else, but I can move it outside to reduce the line difference